src/bos/operators/base.py

```diff
@@ -253,6 +281,33 @@ def _update_database_for_failure(self, components: List[dict]) -> None:
         self.bos_client.components.update_components(data)
 
 
+def _chunk_size(components: List[dict], max_batch_size: int) -> int:
+    """
+    Given the list of components and the max batch size, return the largest subgroup size
+    that can be used to evenly subdivide it, without exceeding the max batch size
```

Comment:
Why evenly subdivide it? Why not just return the maximum batch size every time until the number of nodes gets below that and then just return the remainder? If the point is to get through quickly without overwhelming the system, then we could run right up to the threshold every time.

The calculus ends up being making larger initial requests and a final smaller request. The number of requests would not change. For example, if you have 91 nodes to process and a maximum batch size of 10, you will always make 10 calls. Using your method, you will have nine calls with 9 nodes in them and a final call with 10.
Alternately, you could have nine calls with 10 nodes in them and a final call with only 1 node in it. The second method returns more nodes to service faster. The former method is less taxing on the system.

Reply:
The latter note you made is the main reason -- the general goal of this PR is to "smooth out" the large request spikes, if possible. I agree with you that if the maximum batch size is a size which works, then it should work either way. And obviously the code is simpler if I don't worry about smoothing it out and instead just do it the way you propose. It's actually the way I had written it first, but I changed it to what I have now for the reasons I explained. But I don't feel strongly about it. If you do, it's no problem to change it back. I'll do that.